src/Marvin.Web/HealthChecks/ExampleHealthCheck.cs

```diff
@@ -0,0 +1,23 @@
+﻿using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace Marvin.Web.HealthChecks
+{
+    public class ExampleHealthCheck : IHealthCheck
+    {
+        public Task<HealthCheckResult> CheckHealthAsync(
+            HealthCheckContext context, CancellationToken cancellationToken = default)
+        {
+            var isHealthy = true;
```

Comment:
We could `Match` on this:

```csharp
return isHealthy.Match(
   t => Task.FromResult(HealthCheckResult.Healthy("A healthy result.")),
   f => Task.FromResult(new HealthCheckResult(context.Registration.FailureStatus, "An unhealthy result."))
);
```

> this is the sort of stuff we do with functional programming style, just showing this as an example, it becomes more powerful when done over types

Reply:
I have now updated this, in line with the comment.